File: example_solutions/C/test2.c
long ans(s,e){long r=0;int w=0,d=0,m=0,y=1900;while(y<=e){r+=(w=++w%7)==5&d==12&y>=s;y+=!(d*=++d<30+(2773>>m&1)-(m==1)*(2-(y%4<1&(y%100||y%400<1))))&&!(m*=12>++m);}return r;}

File: example_solutions/python/test2.py
def ans(s,e):
 r=w=d=m=0;y=1900
 while y<=e:r+=(w==4)*(d==12)*(y>=s);w=-~w%7;d=-~d%(31-(m in(3,5,8,10))+(m==1)*((y%4<1)*((y%25>0)|(y%16<1))-3));m+=1>d;m%=12;y+=m+d<1
 return r